core/templates/dev/head/AppInit.ts

```diff
@@ -49,6 +71,15 @@ var oppia = angular.module(
     'toastr', 'ui.bootstrap', 'ui.sortable', 'ui.tree', 'ui.validate',
     downgradedModule
   ].concat(
-  window.GLOBALS ? (window.GLOBALS.ADDITIONAL_ANGULAR_MODULES || []) : []));
+  window.GLOBALS ? (window.GLOBALS.ADDITIONAL_ANGULAR_MODULES || []) : []))
+  .directive(
+    'serviceBootstrap',
```

Comment:
Hm, I largely don't understand what's going on here, and it seems a bit odd to me that this is a directive because I have a mental model of directives as "user-visible html things".

Could you point to a reference that explains this pattern more, please? Thanks!

Reply:
The real reason to use this was to load Angular. AngularJS, when used with downgradeModule loads Angular lazily only when it encounters the first Angular component.
More details [here](https://stackoverflow.com/a/48668591/9121240).

Reply:
Ah ok, I think you should include this information and a description in the code, otherwise devs won't know what's going on.

Also, is the idea to delete this component post-migration? If so, could you please file and self-assign an issue for that?

Reply:
Yes, I think that this would not be needed later when the main component itself would be the bootstrapping call.
Yes, I will open an issue as soon as this gets in.

Reply:
Please go ahead and file the issue now, and link it in this comment thread (you can prefix it with `[Once #xxx gets in]`). That way we can actually resolve all comment threads before we merge this.

Also, make sure to include sufficient documentation/comments in the code about this, I think this is necessary before the PR is merged.

Reply:
I have added the comments PTAL!

Reply:
Created #7072!